knimin/lib/data_access.py

```diff
@@ -1402,6 +1402,60 @@ def getAGKitDetails(self, supplied_kit_id):
         else:
             return {}
 
+    def push_barcode_to_qiita_buffer(self, barcode):
+        """Adds barcode to the qiita buffer
+
+        Parameters
+        ----------
+        barcode : str
+            The identifier to stage
+        """
+        sql = """SELECT barcode
+                 FROM project_qiita_buffer"""
+        present = {i[0] for i in self._con.execute_fetchall(sql)}
```

Comment:
not sure if worth considering but this will fail if there are no barcodes ...

Reply:
perhaps worth replacing for fetchflatten, but not sure if available in code base

Reply:
Sorry, wrong button.  Not sure if fetch flatten is in here, will use if available, thanks!